src/reducers/messagesReducer.js

```diff
@@ -0,0 +1,17 @@
+import * as types from '../actions/actionTypes';
+import initialState from './initialState';
+
+export default function messagesReducer(state = initialState.messages, action) {
+  switch (action.type) {
+    case types.FETCH_MESSAGES_SUCCESS:
+      return Object.keys(action.payload).map(key => {
```

Comment:
In theory, you should merge the payload wit the current state: you add the new messages to the currently loaded messages.

Given the fact you are using the `value` event from Firebase, it does not change anything but, as mentioned above, you should not really rely on `value` only.